server/src/services/course.ts

```diff
@@ -31,22 +33,35 @@ export async function findCourseById(courseId: number): Promise<CourseWithTransl
   return course as CourseWithTranslation;
 }
 
-export async function findAllInstances(courseId: number): Promise<Array<CourseInstance>> {
-  const course: Array<Course> | null = await models.Course.findAll({
-    attributes: ['id'],
-    where: {
-      id: courseId
-    }
-  });
+export interface InstanceWithTeacher {
+  CourseInstance: CourseInstance
+  Teacher: User | null
+}
+
+export async function findAllInstances(courseId: number): Promise<Array<Course | Array<InstanceWithTeacher>>> {
   
-  if (course == null || course.length == 0) throw new Error (`course with id ${courseId} not found`);
+  const course: Course | null = await Course.findByPk(courseId);
+  
+  if (!course) throw new Error (`course with id ${courseId} not found`);  
 
-  const instances: Array<CourseInstance> | null = await models.CourseInstance.findAll({
-    attributes: ['courseId', 'gradingType', 'startingPeriod', 'endingPeriod'],
+  const instances: Array<CourseInstance> = await CourseInstance.findAll({
+    attributes: ['id', 'courseId', 'gradingType', 'startingPeriod', 'endingPeriod', 'teachingMethod', 'responsibleTeacher', 'minCredits', 'maxCredits', 'startDate', 'endDate', 'createdAt', 'updatedAt'],
     where: {
       courseId: courseId
     }
   });
 
-  return instances;
+  const instancesWithTeacher: Array<InstanceWithTeacher> = [];
+
+  if (instances.length != 0) {
+    for (const instance of instances) {
+      const teacher: User | null = await User.findByPk(instance.responsibleTeacher);
```

Comment:
can you use `include` in the `CourseInstance` db query so there is no need for this loop?

Reply:
Sequelize [documentation](https://sequelize.org/docs/v6/core-concepts/assocs/#defining-an-alias) on using alias when there exists more than one association between two models